File: audience/src/main/java/br/com/concretesolutions/audience/core/rule/LoggingShowRule.java
package br.com.concretesolutions.audience.core.rule;

import android.util.Log;

import br.com.concretesolutions.audience.core.MessageEvent;

public final class LoggingShowRule implements AssistantAndScriptRule {

    private static final String TAG = LoggingShowRule.class.getSimpleName();

    @Override
    public boolean shouldIntercept(String assistantKey) {
        return true;
    }

    @Override
    public boolean shouldIntercept(Class<?> messageClass) {
        return true;
    }

    @Override
    public MessageEvent<String> interceptAssistantMessage(MessageEvent<String> message) {
        Log.i(TAG, "Intercepted assistant message: " + message);
        return message;
    }


    @Override
    public MessageEvent<?> interceptScriptMessage(MessageEvent<?> message) {
        Log.i(TAG, "Intercepted script message: " + message);
        return message;
    }
}

File: audience-core/src/main/kotlin/io/github/victorolinasc/audience/core/rule/LoggingShowRule.kt
package io.github.victorolinasc.audience.core.rule

import android.util.Log

import io.github.victorolinasc.audience.core.MessageEvent

class LoggingShowRule : AssistantAndScriptRule {

    override fun shouldIntercept(assistantKey: String): Boolean {
        return true
    }

    override fun shouldIntercept(messageClass: Class<*>): Boolean {
        return true
    }

    override fun interceptAssistantMessage(message: MessageEvent<String>): MessageEvent<String> {
        Log.i(TAG, "Intercepted assistant message: $message")
        return message
    }


    override fun interceptScriptMessage(message: MessageEvent<*>): MessageEvent<*> {
        Log.i(TAG, "Intercepted script message: $message")
        return message
    }

    companion object {

        private val TAG = LoggingShowRule::class.java.simpleName
    }
}
